File: MB-Chat/observability_py/structured_logger.py
from datetime import UTC, datetime
from typing import Any
from uuid import uuid4

from .telemetry_sanitizer import sanitize_telemetry_payload
from .types import ObservabilityEvent, ObservabilityLayer, ObservabilitySeverity, TraceContext


def build_structured_log(
    trace: TraceContext,
    layer: ObservabilityLayer,
    event_type: str,
    severity: ObservabilitySeverity = "info",
    payload_summary: dict[str, Any] | None = None,
    safety_flags: list[str] | None = None,
) -> ObservabilityEvent:
    return ObservabilityEvent(
        event_id=f"evt_{uuid4()}",
        trace_id=trace.trace_id,
        correlation_id=trace.correlation_id,
        layer=layer,
        event_type=event_type,
        severity=severity,
        payload_summary=sanitize_telemetry_payload(payload_summary or {}),
        safety_flags=safety_flags or [],
        created_at=datetime.now(UTC).isoformat(),
    )

File: MB-Secretaria/observability/structured-logger.ts
import { randomUUID } from "node:crypto";
import { sanitizeTelemetryPayload } from "./telemetry-sanitizer";
import type { ObservabilityEvent, ObservabilityLayer, ObservabilitySeverity, TraceContext } from "./types";

export function buildStructuredLog(input: {
  trace: TraceContext;
  layer: ObservabilityLayer;
  event_type: string;
  severity?: ObservabilitySeverity;
  payload_summary?: Record<string, unknown>;
  safety_flags?: string[];
}): ObservabilityEvent {
  return {
    event_id: `evt_${randomUUID()}`,
    trace_id: input.trace.trace_id,
    correlation_id: input.trace.correlation_id,
    layer: input.layer,
    event_type: input.event_type,
    severity: input.severity ?? "info",
    payload_summary: sanitizeTelemetryPayload(input.payload_summary ?? {}),
    safety_flags: input.safety_flags ?? [],
    created_at: new Date().toISOString(),
  };
}
